### ISSUE TITLE
Issues with sample paths (Win10)

### ISSUE THREAD

**Issue by DreadDendy** [stargatedaw/stargate#39]
Hello, I have encountered several issues.
The first of them is similar to #27 and #29.
When I used built-in file browser (Item Editor/Audio) and drag-and-drop sample file from some other project (one of demo projects in this case), I have got a lot of extra directories,
`C:\Users\DreadDendy\stargate\projects\Test\audio\samples\C\Users\DreadDendy\stargate\projects\001-dubstep\audio\samples\home\fd\src\github.com\stargateaudio\stargate-sample-pack\stargate-sample-pack\freesound\drums\cymbal\closed\339278__cabled-mess__hihat-closed-raw-03-gate-eq.wav`
Then I have found **audio_pool** file, which contains references for audio files.
It seems a bit like mess. If DAW make copies of added samples may be it would be more logical to use relative paths, like !/timestretch/4.wav?
```
0|0.0|C:/Users/DreadDendy/stargate/projects/001-dubstep/audio/samples/home/fd/src/github.com/stargateaudio/stargate-sample-pack/stargate-sample-pack/freesound/drums/bongo/99754__menegass__bongo4.wav
1|0.0|C:/Users/DreadDendy/stargate/projects/001-dubstep/audio/samples/home/fd/src/github.com/stargateaudio/stargate-sample-pack/stargate-sample-pack/freesound/drums/cymbal/closed/339278__cabled-mess__hihat-closed-raw-03-gate-eq.wav
2|0.0|C:/Users/DreadDendy/stargate/projects/001-dubstep/audio/samples/home/fd/src/github.com/stargateaudio/stargate-sample-pack/stargate-sample-pack/freesound/drums/cymbal/crash/104217__minorr__crash-zildian-a-custom-16-centr-f.wav
3|0.0|C:/Users/DreadDendy/stargate/projects/001-dubstep/audio/samples/home/fd/src/github.com/stargateaudio/stargate-sample-pack/stargate-sample-pack/freesound/drums/kick/266764__hard3eat__synth-kick-3.wav
4|0.0|!/timestretch/4.wav
5|0.0|C:/Users/DreadDendy/stargate/projects/001-dubstep/audio/samples/home/fd/src/github.com/stargateaudio/stargate-sample-pack/stargate-sample-pack/freesound/drums/snare/212234__alexthegr81__tapesnare-21.wav
6|0.0|/:C/Users/DreadDendy/stargate/projects/001-dubstep/audio/samples/home/fd/src/github.com/stargateaudio/stargate-sample-pack/stargate-sample-pack/freesound/drums/bongo/99754__menegass__bongo4.wav
7|0.0|/:C/Users/DreadDendy/stargate/projects/001-dubstep/audio/samples/home/fd/src/github.com/stargateaudio/stargate-sample-pack/stargate-sample-pack/freesound/drums/bongo/99754__menegass__bongo4.wav
\
```
Then I attempted to re-create project and added samples manually, without DAW.
I have added some folders with files in C:\Users\DreadDendy\stargate\projects\Test\audio\samples.
And then when I have launched Stargate and added samples to track it worked for the first time, but then I relaunched DAW and it was unable to load project.
The error is
`[2023-05-11 12:27:31,838] ERROR sglib\lib\process.py          : 75 - C:\Program Files\stargatedaw@github\Stargate\program\engine\stargate-engine.exe: "Assertion failed: Audio file does not exist: '/:snare/212234__alexthegr81__tapesnare-21.wav'"`
and **audio_pool** file is
```
0|0.0|/:snare/212234__alexthegr81__tapesnare-21.wav
1|0.0|/:snare/212234__alexthegr81__tapesnare-21.wav
2|0.0|/:kick/266764__hard3eat__synth-kick-3.wav
3|0.0|/:kick/266764__hard3eat__synth-kick-3.wav
4|0.0|/:kick/266764__hard3eat__synth-kick-3.wav
5|0.0|/:kick/266764__hard3eat__synth-kick-3.wav
\
```
When I changed it to
```
0|0.0|!/samples/snare/212234__alexthegr81__tapesnare-21.wav
1|0.0|!/samples/snare/212234__alexthegr81__tapesnare-21.wav
2|0.0|!/samples/kick/266764__hard3eat__synth-kick-3.wav
3|0.0|!/samples/kick/266764__hard3eat__synth-kick-3.wav
4|0.0|!/samples/kick/266764__hard3eat__synth-kick-3.wav
5|0.0|!/samples/kick/266764__hard3eat__synth-kick-3.wav
\
```
the project have been successfully opened.

**Comment by L3337** [comment]
Hi @DreadDendy ,

Thanks for reporting.  I'm looking into this, I'll let you know when I have it figured out.

**Comment by L3337** [comment]
I fixed the main problem, which was this:
```
0|0.0|/:snare/212234__alexthegr81__tapesnare-21.wav
1|0.0|/:snare/212234__alexthegr81__tapesnare-21.wav
2|0.0|/:kick/266764__hard3eat__synth-kick-3.wav
3|0.0|/:kick/266764__hard3eat__synth-kick-3.wav
4|0.0|/:kick/266764__hard3eat__synth-kick-3.wav
5|0.0|/:kick/266764__hard3eat__synth-kick-3.wav
\
```

I did not yet try to implement an automatic correction of the problem in existing projects, I am still looking into that.

> If DAW make copies of added samples may be it would be more logical to use relative paths, like !/timestretch/4.wav?

This system that is in place has been there for many years, not easily possible to change it without risking breaking backwards compatibility with existing projects.  In the case of dragging samples from a DIFFERENT project, this is the expected behavior, because the samples came from outside the project.  What should have been working, but was not, was being able to drag samples from the same project in Windows.

>  I have added some folders with files in C:\Users\DreadDendy\stargate\projects\Test\audio\samples.

Be careful, that folder was not meant to be manually updated (although I get you were fixing the project in this case).  The only project folder meant to be manually modified by the user is:

>  C:\Users\DreadDendy\stargate\projects\Test\user

Modifying the `audio` folder by hand (audio or project files) is quite risky if you don't understand how all of the files relate to each other.
